### ISSUE TITLE
Fix grid refreshing

### ISSUE THREAD

**Issue by asmecher** [pkp/pkp-lib#2751]
A recent code change appears to have broken grid refreshing -- an item, when bounced off the server e.g. by creation or edit, will re-appear in the grid but will no longer be editable and may display incorrectly.

**Comment by NateWr** [comment]
I have also noticed that in some cases (eg - Email Templates), when an item is edited, the grid refreshes with that item at the top. And items which would have appeared above it seemed to disappear.

**Comment by defstat** [comment]
It seems that upon refresh, the edited row loses its parent <tr>
For example at the announcements grid, before editing a row the table has, lets say, 2 rows:
<tr id="component-grid-announcements-announcementtypegrid-row-2" class="gridRow has_extras">
<tr id="component-grid-announcements-announcementtypegrid-row-3" class="gridRow has_extras">

After editing the first row, the table has 
<tr id="component-grid-announcements-announcementtypegrid-row-3" class="gridRow has_extras">
and the content of <tr id="component-grid-announcements-announcementtypegrid-row-2" class="gridRow has_extras">, but without the <tr> element as parent of the content. (the <tr id="component-grid-announcements-announcementtypegrid-row-2" class="gridRow has_extras"> element is missing)

That effects the $.pkp.controllers.grid.GridHandler.prototype.resequenceRows function.

I get that when enabling plugins as well.

**Comment by NateWr** [comment]
Thanks @defstat! That's helpful.

This is almost certainly related to the work I had to do to destroy controllers as HTML chunks get replaced (to ensure Vue.js event listeners get de-registered). That's all in `Handler.js` in the `replaceWith`, `replacePartial` and `unbindPartial` methods.

**Comment by asmecher** [comment]
@NateWr, I think I've tracked this down: https://github.com/pkp/pkp-lib/commit/b6a38601bebe601bd3752d1f39348f0204e33439#diff-25cab5d6788e8bf3432c3a755f10c1a4L681 passed the new row through `.html()` to convert to a string, but that has the side-effect of unwrapping it from its containing element. The change appears to have been done to pass the JS linter, as the `.replacePartialWith` func asks for a `string` per its code header.

However, the underlying functions will also accept a `jQueryObject`, so I've reverted the `.html()` wrap and added `jQueryObject` to the list of types accepted by the underlying functions.

Make sense? I'm committing, as it seems fine to me.
